src/main/java/dinkplugin/notifiers/DiaryNotifier.java

```diff
@@ -60,6 +67,39 @@ public void onTick() {
         }
     }
 
+    public void onMessageBox(String message) {
+        if (!isEnabled()) return;
+
+        Matcher matcher = COMPLETION_REGEX.matcher(message);
+        if (matcher.find()) {
+            String difficultyStr = matcher.group("difficulty");
+            AchievementDiary.Difficulty difficulty;
+            try {
+                difficulty = AchievementDiary.Difficulty.valueOf(difficultyStr.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                log.warn("Failed to match diary difficulty: {}", difficultyStr);
+                return;
+            }
+
+            String area = matcher.group("area");
+            Optional<Integer> found = DIARIES.entrySet().stream()
+                .filter(e -> e.getValue().getRight() == difficulty && e.getValue().getLeft().equals(area))
+                .map(Map.Entry::getKey)
+                .findAny();
+            if (found.isPresent()) {
+                int varbitId = found.get();
+                if (isComplete(varbitId, 2)) {
```

Comment:
This will always return true, you'll need to inverse the check to check

Reply:
Good catch